core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -66,6 +66,17 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
   Map("clientId" -> clientId)
   )
 
+  newGauge("DeadFetcherThreadCount", {
+    new Gauge[Int] {
+      def value: Int = {
+        deadFetcherThreadCount
+      }
+    }
+  })
+
+  // visible for testing
+  private[server] def deadFetcherThreadCount = fetcherThreadMap.count(_._2.isShutdownComplete)
```

Comment:
I think the same condition applies here. Since we don't use the `lock` in this method, we may catch an intermediate state where some threads are shut down and not removed from the map. This can happen under common conditions like dynamic broker config updates (e.g `AbstractFetcherManager#resizeThreadPool()`)